packages/common/src/search/Collection.ts

```diff
@@ -0,0 +1,92 @@
+import { hubSearchQuery, IArcGISContext } from "..";
+import {
+  IHubCatalog,
+  IHubCollection,
+  IHubSearchOptions,
+  IHubSearchResponse,
+  IHubSearchResult,
+  IQuery,
+} from "./types";
+
+/**
+ * Implements Collection behavior
+ */
+/* istanbul ignore next */
+export class Collection implements IHubCollection {
+  private _context: IArcGISContext;
+  private _collection: IHubCollection;
+  private constructor(collection: IHubCollection, context?: IArcGISContext) {
+    this._collection = collection;
+    if (context) {
+      this._context = context;
+    }
+  }
+
+  public static async create(
+    collection: IHubCollection,
+    context?: IArcGISContext
+  ): Promise<Collection> {
+    // Async so we could extend to look up by ref like
+    // https://mysite.com#documents vs passing in a full ICollection
```

Comment:
I get this, but I wonder how much better: 

`const collection = await Collection.create(url)`

is than: 

`const collection = Collection.create(await fetchCollection(url))`

The latter ensures that the fetching logic is tree-shakable.

Reply:
This class is not exported, so we can revisit and revise as we move forward